library/include/internal/hipsolver-types.h

```diff
@@ -33,25 +33,71 @@ typedef enum
     HIPSOLVER_STATUS_ZERO_PIVOT        = 12,
 } hipsolverStatus_t;
 
-// set the values of enum constants to be the same as those used in cblas
+#ifndef HIPBLAS_OPERATION_DECLARED
+#define HIPBLAS_OPERATION_DECLARED
```

Comment:
We could do more to make this compatible with older versions of hipBLAS. There are a number of specific cases we could make compatible, although I don't think there's anything that would make it compatible in every case.

For example, when compiling with C++17 or a compiler that supports the extension, we could use `__has_include` and include `<hipblas/hipblas.h>` if available instead of using forward-declarations. Or, we could skip declaring the enums if `hipblasVersionMajor` is defined (since this indicates that the hipblas header has already been included). Those would all extend the source-compatibility of hipSOLVER with older versions of hipBLAS in a variety of cases. 

However, that would certainly be more complex and it could itself introduce more problems for users if we made any mistakes, so I'm fine with leaving this as-is.